pkg/iac/rego/load.go

```diff
@@ -239,13 +239,10 @@ func (s *Scanner) prunePoliciesWithError(compiler *ast.Compiler) error {
 
 func (s *Scanner) compilePolicies(srcFS fs.FS, paths []string) error {
 
-	schemaSet, custom, err := BuildSchemaSetFromPolicies(s.policies, paths, srcFS, s.customSchemas)
+	schemaSet, err := BuildSchemaSetFromPolicies(s.policies, paths, srcFS, s.customSchemas)
 	if err != nil {
 		return err
 	}
-	if custom {
-		s.inputSchema = nil // discard auto detected input schema in favor of check defined schema
-	}
```

Comment:
I'm not sure if I understand, this would be the case if the user passes in their own schema. If we get rid of it, what would happen?

Reply:
`custom` has always been true

Reply:
But what if a check that is not ours, doesn't have a schema?

Reply:
[This](https://github.com/aquasecurity/trivy/pull/7517/files#diff-05bb67fcd5d0722bb3d41fbff6495ec33fa36e009727c001a8c859ebb3d3185cR16) function loads all checks, including ours and custom checks